src/results.rs

```diff
@@ -0,0 +1,278 @@
+                .map_err(|e| PyRuntimeError::new_err(e.to_string()))?;
+
+            let column_name = PyString::new(py, raw_col.spec.name()).unbind();
+            let value = val.into_pyobject(py).expect("Can't fail").unbind();
+
+            Ok(Column { column_name, value })
+        })
+    }
+
+    fn next_row(&mut self) -> PyResult<()> {
+        let column_iterator = self
+            .row_iterator
+            .next()
+            .ok_or_else(|| PyErr::new::<pyo3::exceptions::PyStopIteration, _>(""))?
+            .map_err(|e| PyRuntimeError::new_err(e.to_string()))?;
+
+        self.column_iterator = column_iterator;
+        Ok(())
+    }
+}
+
+// NOTE: In method attach_to_cart we cannot move out cart so We cannot use into_inner inside of it.
+// Fields in DeserializedMetadataAndRawRows are private We have to think
+// If it would be better to use method into_inner and construct new RawMetadata struct
+// or to implement some public method in rust driver that would give access to raw_rows.
+struct RawMetadata {
+    metadata: ResultMetadataHolder,
+    rows_count: usize,
+    raw_rows: Bytes,
+}
+
+impl RawMetadata {
+    fn from_inner(data: DeserializedMetadataAndRawRows) -> RawMetadata {
+        let (metadata, rows_count, raw_rows) = data.into_inner();
+
+        RawMetadata {
+            metadata,
+            rows_count,
+            raw_rows,
+        }
+    }
+}
+
+struct RawMetadataCart(Box<RawMetadata>);
+
+impl Deref for RawMetadataCart {
+    type Target = RawMetadata;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+unsafe impl StableDeref for RawMetadataCart {}
+
+#[pyclass(name = "ColumnIterator")]
+pub struct RowColumnIterator {
+    yoked: Yoke<IteratorWrapper<'static>, RawMetadataCart>,
+}
+
+impl RowColumnIterator {
+    fn new(result: QueryRowsResult) -> Self {
+        let (data, _, _, _) = result.into_inner();
+
+        let cart = RawMetadataCart(Box::new(RawMetadata::from_inner(data)));
+
+        let yoked = Yoke::attach_to_cart(cart, |cart| {
+            let frame_slice = FrameSlice::new(&cart.raw_rows);
+            let row_iterator = RawRowIterator::new(
+                cart.rows_count,
+                cart.metadata.inner().col_specs(),
+                frame_slice,
+            );
+
+            let column_iterator =
+                ColumnIterator::new(cart.metadata.inner().col_specs(), frame_slice);
+
+            IteratorWrapper {
+                row_iterator,
+                column_iterator,
+            }
+        });
+
+        Self { yoked }
+    }
+}
+
+#[pymethods]
+impl RowColumnIterator {
+    pub fn __next__(&mut self) -> PyResult<Column> {
+        self.yoked.with_mut_return(|view| view.next_column())
+    }
+    pub fn __iter__(slf: PyRefMut<'_, Self>) -> PyRefMut<'_, Self> {
+        slf
+    }
+}
+
+#[pyclass]
+pub struct Column {
+    #[pyo3(get)]
+    column_name: Py<PyString>,
+    #[pyo3(get)]
+    value: Py<PyAny>,
```

Comment:
:question: Can we make this typed? Don't we have a union type for all possible Python values deserializable from CQL?